adapters/tlsio_openssl.c

```diff
@@ -970,6 +1001,14 @@ static int create_openssl_instance(TLS_IO_INSTANCE* tlsInstance)
         log_ERR_get_error("unable to set cipher list.");
         result = MU_FAILURE;
     }
+    else if ((tlsInstance->sigalgs_list != NULL) &&
```

Comment:
>sigalgs_list [](start = 27, length = 12)

Is this required? How is it internally handled if we do not pass this in?

Reply:
This is something that is required for our particular hardware (as it for the moment can't sign SHA512 hashes), but may also be required for someone else's. This option enables user to restrict signature algorithms used by OpenSSL. If you don't set it, OpenSSL will use its default list, which is what most users want.

I've made this change a separate commit, I can also make it a separate PR if you believe it's easier to review it this way.